src/System.Slices/src/System/Span.cs

```diff
@@ -283,45 +288,25 @@ public bool ReferenceEquals(Span<T> other)
 
         public override int GetHashCode()
         {
-            // TODO: Write something better
-            uint hash = unchecked((uint)Length);
-            foreach (T el in this)
+            var hashCode = Length ^ Offset.GetHashCode();
```

Comment:
I don't think this is a good hashing function although if someone is using span as a key then they already have a problem :)